Brokerages/InteractiveBrokers/InteractiveBrokersSymbolMapper.cs

```diff
@@ -236,14 +288,12 @@ public static Contract ParseMalformedContractOptionSymbol(Contract malformedCont
         {
             Log.Trace($"InteractiveBrokersSymbolMapper.ParseMalformedContractOptionSymbol(): Parsing malformed contract: {InteractiveBrokersBrokerage.GetContractDescription(malformedContract)} with trading class: \"{malformedContract.TradingClass}\"");
 
-            var contractInfoSplit = malformedContract.Symbol.Substring(malformedContract.Symbol.IndexOf('['))
-                .Replace("[", "")
-                .Replace("]", "")
-                .Split(' ');
+            var matches = Regex.Matches(malformedContract.Symbol, @"^.*[\[]\s*(\S*)\s*(\S*)\s*(\S*)[\]]");
```

Comment:
I suggest adding a comment on what this regexp does, since it's not as easy to follow as the previous one.